### PR TITLE
Actually add CI workflow

### PR DESCRIPTION
**Pull request by Ulthran**
The latest PR to be merged had a few things missing, this should get everything working properly on the CI front

Adds .github/workflows/R-CMD-CHECK.yaml for running R CMD check routinely
Moves package docs previously in R/usedist.R to R/usedist-package.R (Deprecated feature, https://r-pkgs.org/man.html#sec-man-package-doc)
Moves in-function imports to namespace imports and Depends required packages rather than Suggests

### CODE REVIEW

File: NAMESPACE
Comment (kylebittinger) [inline NAMESPACE:1]: We should use the qualified versions of all these functions (e.g. rlang::as_name) in the package code so we don't have to import them here.

File: R/dist.R
Comment (kylebittinger) [inline R/dist.R:1]: We should be able to use stats::as.dist (etc.) and avoid the @importFrom statements.

**Review by kylebittinger — COMMENTED** [review]
So close! Comments on namespace stuff and the .Rprofile.

File: .Rprofile
[the remark this answers is not in the document]
Comment (Ulthran) [reply .Rprofile:1]: This seems to be fairly standard practice from what I've seen, but would also be fine with ignoring it and allowing users more flexibility in their dev setup.

**Pull request merged**